js/content/store.js

```diff
@@ -3007,6 +3008,70 @@ let SearchPageClass = (function(){
+                    </button>
+                    <button class="btnv6_blue_hoverfade btn_medium es_search_action" style="width:100%;text-align:center;margin-bottom:3px;" data-url="/api/removefromwishlist">
+                        <span>${Localization.str.search_results.wishlist_remove}</span>
+                    </button>
+                    <hr class="block_content" style="margin-bottom:3px;">
+                    <button class="btnv6_blue_hoverfade btn_medium es_search_action" style="width:100%;text-align:center;margin-bottom:3px;" data-url="/explore/followgame/">
+                        <span>${Localization.str.search_results.follow}</span>
+                    </button>
+                    <button class="btnv6_blue_hoverfade btn_medium es_search_action" style="width:100%;text-align:center;margin-bottom:3px;" data-url="/explore/followgame/" data-form='{"unfollow":1}'>
+                        <span>${Localization.str.search_results.unfollow}</span>
+                    </button>
+                    <hr class="block_content" style="margin-bottom:3px;">
+                    <button class="btnv6_blue_hoverfade btn_medium es_search_action" style="width:100%;text-align:center;margin-bottom:3px;" data-url="/recommended/ignorerecommendation/">
+                        <span>${Localization.str.search_results.ignore}</span>
+                    </button>
+                    <button class="btnv6_blue_hoverfade btn_medium es_search_action" style="width:100%;text-align:center;margin-bottom:3px;" data-url="/recommended/ignorerecommendation/" data-form='{"remove":1}'>
+                        <span>${Localization.str.search_results.unignore}</span>
+                    </button>
+                </div>
+            </div>`;
+
+        HTML.beforeBegin("#additional_search_options > .block", buttonsHtml);
+        ExtensionLayer.runInPageContext(`function() {
+            $J("button.es_search_action").on("click", function(event) {
+                let appids = $J(".search_result_row[data-ds-appid]:not(:hidden)").get().map(e => parseInt($J(e).data("ds-appid")));
+                let n = appids.length;
+                let btn = $J(event.target).closest("button.es_search_action");
+                let text = btn.text();
+                let prompt = ShowConfirmDialog(text, "${Localization.str.search_results.confirm}".replace("__n__", n));
+                prompt.done(function(result) {
+                    if (result !== "OK") {
+                        return;
+                    }
+
+                    let wait = ShowBlockingWaitDialog(text, "${Localization.str.loading} " + (appids.length - n) + "/" + appids.length);
+                    let url = btn.data("url");
+                    let formData = btn.data("form") || {};
+                    formData.sessionid = g_sessionID;
+                    appids.forEach(function(appid) {
+                        formData.appid = appid;
+                        $J.post(url, formData).always(function() {
```

Comment:
Are you ignoring errors here?

Reply:
Yes, listening only for `done` callbacks may lead to incorrect counting and never-ending progression. I could handle errors, if you think that is necessary.

Reply:
The `always` callback definitely has to stay.
I think handling errors would be good here, since we wanted to improve error handling in general

Reply:
Solved in https://github.com/tfedor/AugmentedSteam/pull/376/commits/6668cbddbe6daf025a0ee44c6a147341eaa52814